config/envs/immutable-kieserver.yaml

```diff
@@ -0,0 +1,213 @@
+              imageChangeParams:
+                automatic: true
+                containerNames:
+                  - "{{.ApplicationName}}-postgresql"
+                from:
+                  kind: ImageStreamTag
+                  namespace: "openshift"
+                  name: "postgresql:10"
+            - type: ConfigChange
+          replicas: 1
+          selector:
+            deploymentConfig: "{{.ApplicationName}}-postgresql-{{$index}}"
+          template:
+            metadata:
+              name: "{{.ApplicationName}}-postgresql-{{$index}}"
+              labels:
+                deploymentConfig: "{{.ApplicationName}}-postgresql-{{$index}}"
+                app: "{{.ApplicationName}}"
+                service: "{{.ApplicationName}}-postgresql-{{$index}}"
+            spec:
+              containers:
+                - name: "{{.ApplicationName}}-postgresql"
+                  image: postgresql
+                  imagePullPolicy: IfNotPresent
+                  livenessProbe:
+                    exec:
+                      command:
+                        - "/usr/libexec/check-container"
+                        - "--live"
+                    initialDelaySeconds: 120
+                    timeoutSeconds: 10
+                  readinessProbe:
+                    exec:
+                      command:
+                        - "/usr/libexec/check-container"
+                    initialDelaySeconds: 5
+                    timeoutSeconds: 1
+                  ports:
+                    - containerPort: 5432
+                      protocol: TCP
+                  volumeMounts:
+                    - mountPath: "/var/lib/pgsql/data"
+                      name: "{{.ApplicationName}}-postgresql-pvol"
+                  env:
+                    - name: POSTGRESQL_USER
+                      value: "rhpam"
+                    - name: POSTGRESQL_PASSWORD
+                      value: "{{.AdminPassword}}"
+                    - name: POSTGRESQL_DATABASE
+                      value: "rhpam7"
+                    - name: POSTGRESQL_MAX_PREPARED_TRANSACTIONS
+                      value: "100"
+              volumes:
+                - name: "{{.ApplicationName}}-postgresql-pvol"
+                  persistentVolumeClaim:
+                    claimName: "{{.ApplicationName}}-postgresql-claim-{{$index}}"
+    ## PostgreSQL deployment config END
+    services:
+      ## PostgreSQL service BEGIN
+      - metadata:
+          annotations:
+            description: The database server's port.
+          labels:
+            application: prod
+            service: "{{.ApplicationName}}-postgresql-{{$index}}"
+          name: "{{.ApplicationName}}-postgresql-{{$index}}"
+        spec:
+          ports:
+            - port: 5432
+              targetPort: 5432
+          selector:
+            deploymentConfig: "{{.ApplicationName}}-postgresql-{{$index}}"
+    ## PostgreSQL service END
+  #{{end}}
+  ## RANGE ends
+  ## KIE Servers END
```

Comment:
Do we need to separate out the console and server for immutable env? My assumption is that we wouldn't, due to changes in the deployment model with the operator. Any initial deployment can be specified, and subsequent KIE servers can just be added to the same CR. Thoughts?